enterprise/internal/own/background/recent_contributors.go

```diff
@@ -73,3 +88,36 @@ func (r *recentContributorsIndexer) indexRepo(ctx context.Context, repoId api.Re
 	commitCounter.Add(float64(len(commitLog)))
 	return nil
 }
+
+func isSubRepoPermsRepo(ctx context.Context, repoID api.RepoID, cache rcache.Cache, checker authz.SubRepoPermissionChecker) (isSubRepoPermsRepo bool, err error) {
+	cacheKey := strconv.Itoa(int(repoID))
+	// Look for the repo in cache to see if we have seen it before instead of hitting the DB.
+	val, ok := cache.Get(cacheKey)
+	if ok {
+		var isSubRepoPermsRepo bool
+		if err := json.Unmarshal(val, &isSubRepoPermsRepo); err != nil {
+			return false, err
+		}
+		return isSubRepoPermsRepo, nil
+	}
+
+	// No entry in cache, so we need to look up whether this is a sub-repo perms repo in the DB.
+	ok, err = authz.SubRepoEnabledForRepoID(ctx, checker, repoID)
```

Comment:
nit: a bit more descriptive naming. we already had `ok` for "found in cache", it may be a bit counter-intuitive to have another `ok` for a completely different thing, wdyt?
```suggestion
	isSubRepoPermsRepo, err = authz.SubRepoEnabledForRepoID(ctx, checker, repoID)
```